restler/utils/logging/ndjson_serializer.py

```diff
@@ -120,6 +120,7 @@ def to_dict(self, omit_request_text=None):
         tags.update(self.tags)
         tags.update(self.sequence_tags)
         request_text = None if omit_request_text == True else self.request
+        request_text = logger.remove_tokens_from_logs(request_text) if Settings().no_tokens_in_logs else request_text
```

Comment:
>Settings().no_tokens_in_logs  [](http://example.com/codeflow?start=71&length=29)

Hi Swamy, thanks for the fix!

The check-in tests did not run because we need to migrate them to GitHub actions (this will be done soon), so you will need to run unit tests locally.  For this change, you can just run 'test_trace_database*' tests for validation.

The Settings() object can't be used directly here - could you please restructure the code as follows, which should fix the issue:

```
    def to_dict(self, omit_request_text=None, remove_tokens_from_logs=True):
        from utils.logger import remove_tokens_from_logs as remove_tokens
        tags = {}
...
        request_text = None if omit_request_text == True else self.request
        request_text = remove_tokens(request_text) if remove_tokens_from_logs else request_text 
```

then just one place in trace_db.py line 202 needs to be updated with the new parameter set to the setting.

Reply:
Thanks for detailed code suggestions.
Somehow I saw the usage of Settings() in the file and assumed it will work.
I did run unit tests locally and they failed, but I ran all of those and it took some time. 
Now the tests are all succeeding. 

Is there a place I should add verification in the unit tests to check if auth tokens are actually getting replaced or not?

Reply:
Hi Swamy, thanks for updating the test.  For validating the token replacement was actually performed, sanity checking for the presence of the OMITTED_TOKEN constant (reading the trace DB file directly without the deserializer) is sufficient. For example, counting the number of requests and making sure there are that many occurrences of this constant).

Reply:
Had to update the unit test to see the impact of the flag and validated that authoization token is getting replaced.